### PR TITLE
Feature/authentication guard

### PR DESCRIPTION
**Pull request by yousiefc**
# Description

This PR includes the following proposed change(s):

- implemented authentication with keycloak via AuthenticationGuard

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

```bash
yarn lint
yarn test -u
```

## Does the change impact or break the Docker build?

- [x] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [x] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/ai-reviewer-admin/.env.template [hunk 1]
```diff
@@ -1,2 +1,5 @@
-REACT_APP_AI_REVIEWER_API_URL=http://localhost:8090/
-PORT=3002
\ No newline at end of file
+PORT = 3002
+REACT_APP_KEYCLOAK_REALM=ai-reviewer
+REACT_APP_KEYCLOAK_CLIENT_ID=ai-reviewer-ui
+REACT_APP_KEYCLOAK_URL=http://localhost:8081/auth
+REACT_APP_AI_REVIEWER_API_URL=http://localhost:8090/
```
Comment (KDNeufeld) [inline src/frontend/ai-reviewer-admin/.env.template:5]: I get this in the Chrome console when I launch the application using docker-compose:
**env-config.js:2 Uncaught SyntaxError: Invalid or unexpected token
asyncToGenerator.js:6 Uncaught (in promise) clientId missing**

If I open up the env-config.js in Chrome, I see this:
image

The reason the PORT=3002 was at the bottom of the file was to fix this issue.  In this commit, the PORT was moved to the top of the list and so there is an extra carriage return after the 3002 and spaces before and after the = is confusing javascript.

File: infrastructure/keycloak/realm-export.json [hunk 2]
```diff
@@ -508,33 +508,6 @@
...
-          "ai-reviewer-api-client"
-        ],
-        "account": [
-          "view-profile",
-          "manage-account"
-        ]
-      },
-      "notBefore": 0,
-      "groups": []
-    },
```
Comment (KDNeufeld) [inline infrastructure/keycloak/realm-export.json:537]: I just want to confirm the removal of this service account.  The ai-reviewer-ui client was added by Adam and used and configured in DEV.  This commit is removing this client id.

If I fix the PORT=3002 issue in the .env.template file, I see this when I launch the application:
image

Notice the url is using client-id=ai-reviewer-ui as a parameter, but it's being removed here and so doesn't exist.

Remember to **docker-compose down** to remove cached containers and then **docker-compose up -d --build** to rebuild them.

Comment (yousiefc) [reply infrastructure/keycloak/realm-export.json:537]: okay I see whats happened, strange that it works for me though. So I had to make the route public for ai-reviewer-ui in order to get it to authenticate (as it is in efiling), however it looks like this action removes service accounts for that client. I'm on it.

Comment (KDNeufeld) [reply infrastructure/keycloak/realm-export.json:537]: Yeah, but I think though that making the ai-reviewer-ui public is the right call.  
We should get Taylor and Adam to chime in.

I read this in the documentation:
https://www.keycloak.org/docs/latest/securing_apps/#_javascript_adapter
_One important thing to note about using client-side applications is that the client has to be a public client as there is no secure way to store client credentials in a client-side application._
(per https://stackoverflow.com/questions/62298549/keycloak-client-secret-not-provided-in-request)

Comment (yousiefc) [reply infrastructure/keycloak/realm-export.json:537]: Thats the documentation I saw as well. Also efiling-demo is configured as public. I'm just waiting for Adam to get back to me on this.

Comment (yousiefc) [reply infrastructure/keycloak/realm-export.json:537]: Got the go ahead from Adam its just going to stay public like efiling

Comment (KDNeufeld) [reply infrastructure/keycloak/realm-export.json:537]: Okay, cool.   I still can't get the app the run locally, but if it works for you then it must be my env.
Awesome work!

File: src/frontend/ai-reviewer-admin/.env.template (hunk 1 above)
Comment (KDNeufeld) [reply src/frontend/ai-reviewer-admin/.env.template]: I committed a small change to fix this error.
Just moving the PORT line to the bottom seems to fix it.

**Review by KDNeufeld — APPROVED** [review]
Awesome work!

**Pull request merged**